### PR TITLE
rpg-battles/sound integration & Wiki

### LINKED ISSUE #1553: sound & rpg-battles: make a design doc for integrating sound effects into rpg-battles module
in this issue, the `sound` team (myself) will work with the `rpg-battles` team to integrate sound effects.

Assignees: `sound`: Victor; `rpg-battles`: Emilio and Nafis

### PR DESCRIPTION
**Pull request by jiaoboxuan**
In this issue, we attempted to integrate sound modules into `rpg-battle` modules. We changed the `CMakeFiles.txt` of both `sound` and `rpg-battle` folders, and added `#include` statements to the `rpg_battle_flow.h` header file. The code compiles successfully, yet the issue is that when we put actual `sound_new`, `play_sound`, and `sound_free` functions, there are some extremely unexpected errors that occur outside of the function where the codes are inserted. We are going to deal with the actual function implementation over the weekend, yet with the successful compilation post addition of header files, we might as well put this pull request in.

Additionally, we wrote a Wiki page on our progress, The link is attached [here](https://github.com/uchicago-cs/chiventure/wiki/Battles-~-Sound-integration-with-battle-(design-document)).

Not sure why this branch cannot automatically merge into `dev`, however. A possibility is that when I created this branch, I copied the files off some branch other than `dev`, so there would be some merging conflicts, but I hope it's not a big issue. Please let us know ASAP if it is indeed a source of concern.

### CODE REVIEW

File: src/sound/src/sound.c [hunk 1]
```diff
@@ -1,6 +1,26 @@
...
+    sound_init(sound, type, name);
+    return sound;
+}
+
+/* See sound.h for details */
+int sound_init(sound_t *sound, SoundType type, char *name)
+{
+    sound->type = type;
+    sound->name = name;
+    if (sound->name == NULL) {
```
Comment (nicholaswlee) [inline src/sound/src/sound.c]: Make sure to follow the style guidelines, use
if ()
{
}

File: include/sound/sound.h [hunk 2]
```diff
@@ -31,12 +31,34 @@ typedef struct {
...
+ *      type: enum SoundType
+ *      name: name of the sound, 
+ *            or more specifically, the path to a sound file.
+ *            Must be already allocated in the heap
+ * Return:
+ *      An allocated sound structure, with name and type
+ */
+sound_t *sound_new(SoundType type, char *name);
+
+/* Initializes the
```
Comment (nicholaswlee) [inline include/sound/sound.h]: I'm assuming this is initializing sound?

**Review by nicholaswlee — CHANGES_REQUESTED** [review]
This pull request looks good. Everything has been defined well and given good header comments. I've left a couple comments for things that are just minor styling issues. You've also done a good job freeing these structs as well. I am excited for sound to get implemented, especially within the scope of battles!

File: include/sound/sound.h (hunk 2 above)
Comment (jwschroeder21) [reply include/sound/sound.h]: yep

**Review by jwschroeder21 — APPROVED** [review]
Aside from Nicholas' style suggestions, everything looks good to me!

File: include/sound/sound.h (hunk 2 above)
Comment (ecoble) [inline include/sound/sound.h]: Please update this header comment with the full sentence, also I believe these changes have already been made in a previous PR.

**Review by ecoble — COMMENTED** [review]
I believe a lot of these changes from the sound team have already been merged into dev, so we will see them removed once this PR gets updated. Looks good overall, you might want to update on this PR that the compilation errors were fixed. Additionally, please fix the merge conflicts.

**Review by ecoble — APPROVED** [review]

**Review by nicholaswlee — APPROVED** [review]

**Comment by emiliolevinspage** [comment]
We are closing this issue because the merge is complete! The integration has been completed so that in the future it will be easy to add sounds to any aspect of battles. The process for doing this is documented in the wiki page referenced above.

**Comment by ecoble** [comment]
Issue Score: Excellent

Comments:
Nice job laying the groundwork for integration, and great job reviewing the PR!

**Pull request merged**